euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/Dataset.java

```diff
@@ -20,17 +20,20 @@
 import cz.seznam.euphoria.core.client.io.DataSink;
 import cz.seznam.euphoria.core.client.io.DataSource;
 import cz.seznam.euphoria.core.client.operator.Operator;
+import cz.seznam.euphoria.core.client.operator.hint.OutputHint;
+import cz.seznam.euphoria.core.client.operator.hint.OutputHintAware;
+
+import javax.annotation.Nullable;
 import java.io.Serializable;
 import java.util.Collection;
-import javax.annotation.Nullable;
 
 /**
  * A dataset abstraction.
  *
  * @param <T> type of elements of this data set
  */
 @Audience(Audience.Type.CLIENT)
-public interface Dataset<T> extends Serializable {
+public interface Dataset<T> extends OutputHintAware<OutputHint>, Serializable {
```

Comment:
I don't understand purpose of this interface. It looks odd in the context of client API. We should definitely keep this set on operator, not dataset, because operator is not (directly) part of client API.

Reply:
interface removed. Hints are now on operator